frame/staking/src/lib.rs

```diff
@@ -1741,8 +1882,23 @@ decl_module! {
 		/// # <weight>
 		/// - Time complexity: at most O(MaxNominatorRewardedPerValidator).
 		/// - Contains a limited number of reads and writes.
+		/// -----------
+		/// N is the Number of payouts for the validator (including the validator)
+		/// Base Weight: 110 + 54.2 * N µs (Median Slopes)
+		/// DB Weight:
+		/// - Read: EraElectionStatus, CurrentEra, HistoryDepth, MigrateEra, ErasValidatorReward,
+		///         ErasStakersClipped, ErasRewardPoints, ErasValidatorPrefs (8 items)
+		/// - Read Each: Bonded, Ledger, Payee, Locks, System Account (5 items)
+		/// - Write Each: System Account, Locks, Ledger (3 items)
+		// TODO: Remove read on Migrate Era
 		/// # </weight>
-		#[weight = 500_000_000]
+		#[weight =
+			110 * WEIGHT_PER_MICROS
+			+ 54 * WEIGHT_PER_MICROS * Weight::from(T::MaxNominatorRewardedPerValidator::get())
+			+ T::DbWeight::get().reads(8)
+			+ T::DbWeight::get().reads(5)  * Weight::from(T::MaxNominatorRewardedPerValidator::get() + 1)
+			+ T::DbWeight::get().writes(3) * Weight::from(T::MaxNominatorRewardedPerValidator::get() + 1)
+		]
```

Comment:
shouldn't be saturating? or since there is no user input we assume safety.

Reply:
as you mentioned, there is no user input. but i am happy to fix

Reply:
I added it to the same list to double check. No need to do at now since it makes things really un-readable.